numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -354,6 +354,10 @@ string_find_loop(PyArrayMethod_Context *context,
         Buffer<enc> buf1(in1, elsize1);
         Buffer<enc> buf2(in2, elsize2);
         npy_intp idx = string_find<enc>(buf1, buf2, *(npy_int64 *)in3, *(npy_int64 *)in4);
+        if (raise_error && idx == -1) {
+            PyErr_SetString(PyExc_ValueError, "substring not found");
```

Comment:
Are we allowed to raise inside the loop? Don't you need `npy_gil_error` here too?